scripts/2015_nips_paper/plot/plot_ranks.py

```diff
@@ -0,0 +1,176 @@
+                # collect all csv files of different seeds for current model and
+                # current task.
+                if model in ['vanilla', 'ensemble']:
+                    csv_file = os.path.join(working_directory,
+                                            'vanilla',
+                                            seed,
+                                            task_id,
+                                            "score_{}.csv".format(model)
+                                            )
+
+                elif model in ['metalearning', 'meta_ensemble']:
+                    csv_file = os.path.join(working_directory,
+                                            'metalearning',
+                                            seed,
+                                            task_id,
+                                            "score_{}.csv".format(model),
+                                            )
+                csv_files.append(csv_file)
+
+            performance_list = []
+            time_list = []
+
+            # Get data from csv
+            for fl in csv_files:
+                _, csv_data = read_csv(fl, has_header=True)
+                csv_data = np.array(csv_data)
+                # Replace too high values with args.maxsize
+                data = [min([sys.maxsize, float(i.strip())]) for i in
+                        csv_data[:, 2]]  # test trajectories are stored in third column
+
+                time_steps = [float(i.strip()) for i in csv_data[:, 0]]
+                assert time_steps[0] == 0
+
+                performance_list.append(data)
+                time_list.append(time_steps)
+
+            # trajectory is the pd.Series object containing all seed runs of the
+            # current model and current task.
+            trajectory = fill_trajectory(performance_list, time_list)
+            trajectories.append(trajectory)
+
+        # list[list[pd.Series]]
+        trajectories_all_models.append(trajectories)
+
+    # Maps from task_id to model to pd.Series object
+    # containing trajectories across different seeds.
+    trajectories_by_task = {t: {m: trajectories_all_models[i][j]
+                                for i, m in enumerate(model_list)}
+                            for j, t in enumerate(task_list)}
+
+    # Step 2. Compute average ranks of the trajectories.
+    #####################################################################################
+    all_trajectories = []
+    for model in model_list:
+        trajectories = [
+            trajectories_by_task[task][model] for task in
+            trajectories_by_task
+        ]
+        all_trajectories.append(trajectories)
+
+    all_rankings = []
+    n_iter = 500  # number of bootstrap samples to use for estimating the ranks.
+    n_tasks = len(trajectories_by_task)
+    for i in range(n_iter):
+        pick = np.random.choice(all_trajectories[0][0].shape[1],
+                                size=(len(model_list, )))

```

Comment:
Could you please explain the comma after model_list? Shouldn't it be `size=(len(model_list), )`?

Reply:
From the code you sent me, the comma was there and I did not change it as size=(len(model_list, )) does the same thing as size=len(model_list). I believe that size should simply be a scalar (4, specifically), and therefore it could be just len(model_list) without any comma. Would you agree?

Reply:
Yes, could you then please remove all commas from that argument?